src/hoc/withPopup.js

```diff
@@ -1,8 +1,64 @@
 import React from 'react';
 
-const withPopup = () => {
-    return () => <section>withPopup()</section>;
-}
+const withPopup =
+  ({ overlayStyles = {}, asideStyles = {}, buttonStyles = {} } = {}) =>
+  (WrappedComponent) => {
+    return class extends React.Component {
+      render() {
+        const defaultOverlayStyles = {
```

Comment:
Pewnie dałoby się ładniej te style zrobić, ale to tylko zadanie ;P (styled + theme)

Reply:
Tak tak, dokładnie. Kwestię z przekazywaniem stylów robiłem na szybko :D